lib/platform.c

```diff
@@ -572,25 +572,25 @@ void STDCALL sf_get_uniq_tmp_dir(char * tmpDir)
   uuid4_generate(uuid_cstr);
 #ifdef _WIN32
   GetTempPath(MAX_PATH, tmpDir);
-  strcat(tmpDir, "\\snowflakeTmp\\");
+  sb_strcat(tmpDir, MAX_PATH, "\\snowflakeTmp\\");
   //sf_create_directory does not recursively create dirs in the path.
   sf_create_directory_if_not_exists(tmpDir);
-  sprintf(tmpDir+strlen(tmpDir),"%s\\",uuid_cstr); 
+  sb_sprintf(tmpDir+strlen(tmpDir), MAX_PATH-strlen(tmpDir), "%s\\", uuid_cstr);
 #else
   const char * tmpEnv = getenv("TMP") ? getenv("TMP") : getenv("TEMP");
 
   if (!tmpEnv)
   {
-    strcat(tmpDir, "/tmp/snowflakeTmp/");
+    sb_strcat(tmpDir, MAX_PATH, "/tmp/snowflakeTmp/");
     //sf_create_directory does not recursively create dirs in the path.
     sf_create_directory_if_not_exists(tmpDir);
-    sprintf(tmpDir,"%s/%s/",tmpDir, uuid_cstr);
+    sb_sprintf(tmpDir, MAX_PATH, "%s/%s/",tmpDir, uuid_cstr);
```

Comment:
This line is going to cause issue as vsnprintf() called internally cannot handle the case where the args contains the output destination.
Instead of generating (tmpDir)/(uuid_cstring), it would generate (tmpDir)/(uuid_cstring).

This would lead to the test_simple_put failure on Linux.